parser.go

```diff
@@ -828,6 +851,26 @@ func (p *Parser) parseColumnOption(ctx *parseCtx, col model.TableColumn, f int)
 				l := model.NewLength(tlen)
 				l.SetDecimal(tscale)
 				col.SetLength(l)
+			} else if check(coloptValues) {
```

Comment:
I think Set/Enum should be separated. If you feel like writing the same code twice is not a good thing to do, consider using a helper method:

```go
ctx.parseSetOrEnum( col.SetEnum )
ctx.parseSetOrEnum( col.SetSet ) // ugh. bad name...

func (ctx *parseCtx) parseSetOrEnum(setter func([]string)) error {
  var values []string
OUTER:
  for {
    ctx.skipWhiteSpaces()
    v := ctx.next()
    if v.Type == SINGLE_QUOTE_IDENT || v.Type == DOUBLE_QUOTE_IDENT {
      values = append(values, v.Value)
    } else {
      return newParseError(ctx, t, "expected RPAREN, SINGLE_QUOTE_IDENT, DOUBLE_QUOTE_IDENT(enum values): %s", v.Type)
    }
    ctx.skipWhiteSpaces()

    switch t := ctx.next(); t.Type {
    case COMMA:
    case RPAREN:
      break OUTER
    default:
      return newParseError(ctx, t, "expected COMMA")
     }
  }
  setter(enumValues)
  return nil
}
```

Reply:
I agree